### PR TITLE
[pkgconf] Update to 1.8 (and a few extras; world rebuild)

### PR DESCRIPTION
**Pull request by Neumann-A**
extras:
- add option `NO_PKG_CONFIG` to `vcpkg_configure_meson` and skip the pkg-config setup
- move `vcpkg_find_acquire_program(PKGCONFIG)` in `vcpkg_fixup_pkgconfig` into the check where it is actually used

Result: `pkgconf` can be built without requiring pkg-config from msys. So vcpkg can theoretically migrate to use pkgconf if it wants to do that.

### CODE REVIEW

File: ports/pkgconf/portfile.cmake [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
 vcpkg_install_meson()
-vcpkg_fixup_pkgconfig()
+vcpkg_fixup_pkgconfig(SKIP_CHECK)
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
```
Comment (JackBoosY) [inline ports/pkgconf/portfile.cmake:20]:
```suggestion
if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")

```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/pkgconf/portfile.cmake (hunk 1 above)
Comment (Neumann-A) [reply ports/pkgconf/portfile.cmake]: done

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
cppkafka:x64-osx:
```
[12/25] /Library/Developer/CommandLineTools/usr/bin/c++ -DCPPKAFKA_STATIC=1 -I/Users/vagrant/Data/buildtrees/cppkafka/src/v0.3.1-3b47aeb285.clean/src/../include/cppkafka -I/Users/vagrant/Data/buildtrees/cppkafka/src/v0.3.1-3b47aeb285.clean/include -isystem /Users/vagrant/Data/installed/x64-osx/include -fPIC -std=c++11 -Wall -g -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk -MD -MT src/CMakeFiles/cppkafka.dir/configuration.cpp.o -MF src/CMakeFiles/cppkafka.dir/configuration.cpp.o.d -o src/CMakeFiles/cppkafka.dir/configuration.cpp.o -c /Users/vagrant/Data/buildtrees/cppkafka/src/v0.3.1-3b47aeb285.clean/src/configuration.cpp
FAILED: src/CMakeFiles/cppkafka.dir/configuration.cpp.o 
/Library/Developer/CommandLineTools/usr/bin/c++ -DCPPKAFKA_STATIC=1 -I/Users/vagrant/Data/buildtrees/cppkafka/src/v0.3.1-3b47aeb285.clean/src/../include/cppkafka -I/Users/vagrant/Data/buildtrees/cppkafka/src/v0.3.1-3b47aeb285.clean/include -isystem /Users/vagrant/Data/installed/x64-osx/include -fPIC -std=c++11 -Wall -g -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk -MD -MT src/CMakeFiles/cppkafka.dir/configuration.cpp.o -MF src/CMakeFiles/cppkafka.dir/configuration.cpp.o.d -o src/CMakeFiles/cppkafka.dir/configuration.cpp.o -c /Users/vagrant/Data/buildtrees/cppkafka/src/v0.3.1-3b47aeb285.clean/src/configuration.cpp
/Users/vagrant/Data/buildtrees/cppkafka/src/v0.3.1-3b47aeb285.clean/src/configuration.cpp:153:5: error: use of undeclared identifier 'rd_kafka_conf_set_offset_commit_cb'; did you mean 'rd_kafka_conf_set_closesocket_cb'?
    rd_kafka_conf_set_offset_commit_cb(handle_.get(), &offset_commit_callback_proxy);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    rd_kafka_conf_set_closesocket_cb
/Users/vagrant/Data/installed/x64-osx/include/librdkafka/rdkafka.h:2057:1: note: 'rd_kafka_conf_set_closesocket_cb' declared here
rd_kafka_conf_set_closesocket_cb (rd_kafka_conf_t *conf,
^
/Users/vagrant/Data/buildtrees/cppkafka/src/v0.3.1-3b47aeb285.clean/src/configuration.cpp:153:55: error: cannot initialize a parameter of type 'int (*)(int, void *)' with an rvalue of type 'void (*)(rd_kafka_t *, rd_kafka_resp_err_t, rd_kafka_topic_partition_list_t *, void *)' (aka 'void (*)(rd_kafka_s *, rd_kafka_resp_err_t, rd_kafka_topic_partition_list_s *, void *)'): different number of parameters (2 vs 4)
    rd_kafka_conf_set_offset_commit_cb(handle_.get(), &offset_commit_callback_proxy);
                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/vagrant/Data/installed/x64-osx/include/librdkafka/rdkafka.h:2058:41: note: passing argument to parameter 'closesocket_cb' here
                                  int (*closesocket_cb) (int sockfd,
                                        ^
2 errors generated.
```

Not related to this changes.

**Comment by BillyONeal** [comment]
This is ready to land in the next rollup.

**Comment by BillyONeal** [comment]
Thanks for your help!

**Pull request merged**